commons/src/main/java/io/aiven/kafka/tiered/storage/commons/chunkmanager/cache/UnboundInMemoryChunkCache.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tiered.storage.commons.chunkmanager.cache;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Base64;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ThreadLocalRandom;
+
+import io.aiven.kafka.tiered.storage.commons.chunkmanager.ChunkKey;
+
+/**
+ * An unbound memory-backed chunk cache.
+ *
+ * <p>This is not suitable for production use, but is OK for testing.
+ */
+public class UnboundInMemoryChunkCache implements ChunkCache {
+    private final ConcurrentHashMap<ChunkKey, byte[]> permanentStorage = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, byte[]> tempStorage = new ConcurrentHashMap<>();
+
+    @Override
+    public Optional<InputStream> get(final ChunkKey chunkKey) throws IOException {
+        Objects.requireNonNull(chunkKey, "chunkKey cannot be null");
+
+        final byte[] bytes = permanentStorage.get(chunkKey);
+        if (bytes == null) {
```

Comment:
Is it actually possible(or expected) to have keys with `null` value in this map? At least before addition there is a check for nullability

Reply:
`ConcurrentHashMap` returns `null` to indicate "key not found"